src/device/mod.rs

```diff
@@ -2037,6 +2053,8 @@ impl TaskRuntime for Runtime {
                                     .await?;
 
                                 direct_entities.session_keeper.remove_node(&public_key).await?;
+
+                                direct_entities.upgrade_sync.downgrade(&public_key).await?;
```

Comment:
It it necessary to use `upgrade_sync` for that? Couldn't you reuse `AnalyticsEvent::from_event` also here?